src/model/include/iDynTree/Model/Indeces.h

```diff
@@ -8,30 +8,10 @@
 #ifndef IDYNTREE_INDECES_H
 #define IDYNTREE_INDECES_H
 
-#include <string>
+#ifdef __DEPRECATED
```

Comment:
Do you have any pointer to the compiler documentation regarding this macro? 
I only found this old forum thread https://bytes.com/topic/c/answers/701064-__deprecated , and at least in Visual Studio ( https://docs.microsoft.com/it-it/cpp/preprocessor/predefined-macros ) it is not defined (I think it is ok to just support this feature in the compiler in which it is easy to support it).